### ISSUE TITLE
Class labeling

### ISSUE THREAD

**Issue by sarath0993** [yassouali/pytorch_segmentation#75]
I have few questions, Please answer me I'm stuck 

1. I have only **one class + background**(not required to get trained), So if it's only one class what would be the num_classes?
Note: I tried to set it 1 then 2, After an epoch loss goes to 0.00 and blank images visualized at tensorboard

2. I set the configuration like below, is that correct?
`ignore_label = 255
ID_TO_TRAINID = {-1: ignore_label, 0: ignore_label, 1: 0}`

In my case, pixel value 0 is background and need not to be trained so I have flagged it as ignore_label and pixel value 1 need to be trained so I set it as 0. But nothing worked eventually..

Please advice/suggest configuration to train one class. I am using cityscapes loader and labeled appropriately
Here is my segmented image sample(1024x1024)
[imgur](https://imgur.com/a/8l4G5Cn)

**Comment by sarath0993** [comment]
By printing through `np.unique(label)` code, I have got some results looks like below
seg_image 1: [0 1]
seg_image 2: [1]
seg_image 3: [0 1]
.
.
.

**Comment by yassouali** [comment]
Hi @sarath0993 

If you train the model on only one class, the model will only predict this class - the learning problem, in this case, is quite easy for the model, simply predict the same class for every pixel, and the loss will go down to zero very quickly.

Better train with the background too, so no need to map the background to ignore_label, you can use both with num_calsses = 2. Also, use a lost of data augmentations to help a bit more. Hope this helps.

**Comment by sarath0993** [comment]
Hi @yassouali 
Thanks for your reply, I will train with class and background as you suggested. I will write the results after completing training of some iterations

**Comment by yassouali** [comment]
You can reopen if you have any questions.

**Comment by sarath0993** [comment]
Hi @yassouali , Training on Cityscapes doesn't work for me. Also my custom dataset is quite similar to Ade dataset, so I have switched to ADE20K. Currently training and evaluating my data with different models, simply trying to get better pixel accuracy.
I have one different problem, inference takes more time when compared to validation time(during training), image count are exactly same. Any suggestion to improve speed of inference?

**Comment by yassouali** [comment]
Hi, @sarath0993. Yes inference is quite slow because it is a multi-scale inference, each image is scaled into five sizes, and the results are combined, so in reality, the inference process is done over 5 times the amount of data. One way to reduce the time is to reduce the number of scales we are testing on (`scales` in inference.py).

**Comment by sarath0993** [comment]
Thank you! Sure I will do that
